solr/core/src/java/org/apache/solr/update/DirectUpdateHandler2.java

```diff
@@ -952,12 +941,15 @@ public void split(SplitIndexCommand cmd) throws IOException {
    *
    * @param cmd - cmd apply to IndexWriter
    * @param writer - IndexWriter to use
-   * @param updateTerm - used if this cmd results in calling {@link IndexWriter#updateDocument}
    */
-  private void updateDocOrDocValues(AddUpdateCommand cmd, IndexWriter writer, Term updateTerm) throws IOException {
+  private Term updateDocOrDocValues(AddUpdateCommand cmd, IndexWriter writer) throws IOException {
     assert null != cmd;
     final SchemaField uniqueKeyField = cmd.req.getSchema().getUniqueKeyField();
     final String uniqueKeyFieldName = null == uniqueKeyField ? null : uniqueKeyField.getName();
+    List<SolrInputDocument> docs = cmd.computeFinalFlattenedSolrDocs();
```

Comment:
I don't think these lines you've added should be placed here, as it's computing final flattened docs when they may not even be needed -- which is when cmd.isInPlaceUpdate.  You could move this entirely to updateDocument()?

Reply:
Yeah, but that means we wouldn't have a way to figure out if the update is a block, which is a parameter which needs to be passed to getIdTerm.
I'm not as familiar with the internals, would it always be false during an inPlace update?

Reply:
If i pass isBlock as false in case of an inPlaceUpdate to getIdTerm, the overwrite test seems to fail, as one of the documents is not overwritten.